shell-switcher.el

```diff
@@ -305,7 +305,8 @@ default) or the other window (if OTHER-WINDOW is t)."
     (when (-in-shell-buffer-p)
       (rswitcher-switch-full -ring))
     (-display-shell-buffer other-window)
-    (-prepare-for-fast-key)))
```

Comment:
You're right, we do lose the ability to create new buffers.  My mistake.  I'll work on a fix.